orangecontrib/prototypes/widgets/contingency_table.py

```diff
@@ -0,0 +1,365 @@
+        self.horizontalHeader().hide()
+        self.verticalHeader().hide()
+        self.horizontalHeader().setMinimumSectionSize(60)
+        self.selectionModel().selectionChanged.connect(self.parent._invalidate)
+        self.setShowGrid(False)
+        self.setSizePolicy(QSizePolicy.MinimumExpanding,
+                           QSizePolicy.MinimumExpanding)
+        self.clicked.connect(self._cell_clicked)
+
+    def _cell_clicked(self, model_index):
+        """Handle cell click event"""
+        i, j = model_index.row(), model_index.column()
+        if not i or not j:
+            return
+        n = self.tablemodel.rowCount()
+        m = self.tablemodel.columnCount()
+        index = self.tablemodel.index
+        selection = None
+        if i == j == 1 or not self.circles and i == n - 1 and j == m - 1:
+            selection = QItemSelection(index(2, 2), index(n - 1, m - 1))
+        elif i == 1 or not self.circles and i == n - 1:
+            selection = QItemSelection(index(2, j), index(n - 1, j))
+        elif j == 1 or not self.circles and j == m - 1:
+            selection = QItemSelection(index(i, 2), index(i, m - 1))
+
+        if selection is not None:
+            self.selectionModel().select(
+                selection, QItemSelectionModel.ClearAndSelect)
+
+    def _item(self, i, j):
+        return self.tablemodel.item(i, j) or QStandardItem()
+
+    def _set_item(self, i, j, item):
+        self.tablemodel.setItem(i, j, item)
+
+    def set_variables(self, variablev, variableh, **kwargs):
+        """
+        Sets class headers and top headers and initializes table structure.
+
+        Parameters
+        ----------
+        variablev : Orange.data.variable.DiscreteVariable
+            Class headers are set to ``variablev.values``, top header is set to ``variablev.name``.
+        variableh : Orange.data.variable.DiscreteVariable
+            Class headers are set to ``variableh.values``, top header is set to ``variableh.name``.
+        """
+        self.classesv = variablev.values
+        self.classesh = variableh.values
+        self.headerv = variablev.name
+        self.headerh = variableh.name
+        self.initialize(**kwargs)
+
+    def set_headers(self, classesv, classesh, headerv=None, headerh=None, **kwargs):
+        """
+        Sets class headers and top headers and initializes table structure.
+
+        Parameters
+        ----------
+        classesv : :obj:`list` of :obj:`str`
+            Vertical class headers.
+        classesh : :obj:`list` of :obj:`str`
+            Horizontal class headers.
+        headerv : :obj:`str`, optional
+            Vertical top header.
+        headerh : :obj:`str`, optional
+            Horizontal top header.
+        """
+        self.classesv = classesv
+        self.classesh = classesh
+        self.headerv = headerv
+        self.headerh = headerh
+        self.initialize(**kwargs)
+
+    def initialize(self, circles=False):
+        """
```

Comment:
I would make this method private, set_variables and set_headers should be used by outside callers.

Reply:
For circles parameter, i would rather add setShowCircles method that changes the value and then call initialize.

Reply:
Does the contingencyTable allow display of circles? If not, should it?